File: Low Level Design (LLD)/DesignPatterns/2. Behavariol/Flyweight/Cpp/main.cpp
#include <iostream>
#include <cstdlib>
#include <ctime>
#include "./include/ShapeFactory.h"

std::string getRandomColor() {
    std::string colors[] = {"Red", "Green", "Blue", "White", "Black"};
    return colors[std::rand() % 5];
}

int getRandomX() {
    return std::rand() % 100;
}

int getRandomY() {
    return std::rand() % 100;
}

int main() {
    std::srand(std::time(nullptr));

    for (int i = 0; i < 5; ++i) {
        std::string color = getRandomColor();
        Shape* circle = ShapeFactory::getCircle(color);
        circle->draw(getRandomX(), getRandomY(), 100);
    }

    return 0;
}

File: Low Level Design (LLD)/DesignPatterns/2. Behavariol/Flyweight/Java/Main.java
public class Main {
    private static final String[] colors = { "Red", "Green", "Blue", "White", "Black" };

    public static void main(String[] args) {
        for (int i = 0; i < 5; ++i) {
            String color = getRandomColor();
            Shape circle = ShapeFactory.getCircle(color);
            circle.draw(getRandomX(), getRandomY(), 100);
        }
    }

    private static String getRandomColor() {
        return colors[(int)(Math.random() * colors.length)];
    }

    private static int getRandomX() {
        return (int)(Math.random() * 100);
    }

    private static int getRandomY() {
        return (int)(Math.random() * 100);
    }
}
